src/Text/Layout/Table/LineStyle.hs

```diff
@@ -80,88 +94,197 @@ unicodeVertical HeavyDash4Line = "┋"
 unicodeVertical Dash2Line      = "╎"
 unicodeVertical HeavyDash2Line = "╏"
 
--- | Unicode representations for top lines, with appropriate joins.
-unicodeTopJoins :: LineStyle -> LineStyle -> String
-unicodeTopJoins = unicodeTopJoins' `on` joinType
-
--- | Unicode representations for bottom lines, with appropriate joins.
-unicodeBottomJoins :: LineStyle -> LineStyle -> String
-unicodeBottomJoins = unicodeBottomJoins' `on` joinType
-
--- | Unicode representations for left lines, with appropriate joins.
-unicodeLeftJoins :: LineStyle -> LineStyle -> String
-unicodeLeftJoins = unicodeLeftJoins' `on` joinType
+-- | Unicode interior joins, specifying the horizontal and vertical lines.
+unicodeJoins :: LineStyle -> LineStyle -> String
+unicodeJoins h v = unicodeJoins4 h h v v
 
--- | Unicode representations for right lines, with appropriate joins.
-unicodeRightJoins :: LineStyle -> LineStyle -> String
-unicodeRightJoins = unicodeRightJoins' `on` joinType
+-- | Unicode interior joins, allowing the lines to change when passing through the vertex.
+-- The argument order is west, east, north, then south.
+unicodeJoins4 :: LineStyle -> LineStyle -> LineStyle -> LineStyle -> String
+unicodeJoins4 w e n s = pure $ uJoins (joinType w) (joinType e) (joinType n) (joinType s)
```

Comment:
These functions determine a join character.  Perhaps it would make sense to include that in the name.  Because if I see the term `join` it is not obvious to me that it is a character.

Reply:
I'm happy to change the name, but it is technically a `String` rather than a `Char`. Are you okay with the longer name `unicodeJoinString`?

Reply:
That's fine.